### ISSUE TITLE
Callable assigned to field in enum is not treated as an enumerated type

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5721]
In the bizarro world of enums where nothing works as you would expect and everything is a special case, it appears that callables are treated specially.

```python
import enum

class MyEnum(enum.Enum):
    a = 1
    b = lambda x: x

print(len(MyEnum))  # 1
print(MyEnum.a)  # MyEnum.a
print(MyEnum.b)  # lambda
```

This is related to mypy issue https://github.com/python/mypy/issues/15540.

**Comment by AlexWaygood** [comment]
(Note though that if you were to wrap the `lambda` in `functools.partial`, it would become an enum member, for... Reasons)

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
